test/e2e/framework/framework.go

```diff
@@ -562,3 +563,9 @@ func NestedString(obj map[string]interface{}, fields ...string) string {
 
 	return str
 }
+
+func DetectProvider(cluster ClusterIndex, nodeName string) string {
+	node, _ := KubeClients[cluster].CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
```

Comment:
We really shouldn't ignore errors.

```suggestion
	node, err := KubeClients[cluster].CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
	Expect(err).NotTo(HaveOccurred())
```

Reply:
You're right.
Will fix it.

Reply:
Added